tracer/test/Datadog.Trace.IntegrationTests/Tagging/AASTagsTests.cs

```diff
@@ -0,0 +1,249 @@
+    {
+        var source = GetNullMockVariables();
+        var settings = new TracerSettings(source);
+        var agentWriter = new AgentWriter(_testApi, statsAggregator: null, statsd: null, spanSampler: null);
+        var tracer = new Tracer(settings, agentWriter, sampler: null, scopeManager: null, statsd: null);
+
+        using (tracer.StartActiveInternal("root"))
+        {
+        }
+
+        await tracer.FlushAsync();
+        var traceChunks = _testApi.Wait();
+        var deserializedSpan = traceChunks.Single().Single();
+        AssertNoTags(deserializedSpan);
+    }
+
+    [Fact]
+    public async Task AasTagsShouldBeSerializedOnLocalRootSpans()
+    {
+        // We are going to create 2 chunks
+        // Both chunks should contain all AAS tags on local root span
+
+        // The first chunk will contain multiple spans
+        // Each non local root spans should contain only aas.site.name and aas.site.type tags
+
+        var source = GetMockVariables();
+        var settings = new TracerSettings(source);
+        var agentWriter = new AgentWriter(_testApi, statsAggregator: null, statsd: null, spanSampler: null);
+        var tracer = new Tracer(settings, agentWriter, sampler: null, scopeManager: null, statsd: null);
+
+        ISpan span1;
+        ISpan span11;
+        ISpan span12;
+        ISpan span121;
+
+        using (var scope1 = tracer.StartActive("1"))
+        {
+            span1 = scope1.Span;
+
+            var traceContext = ((Scope)scope1).Span.Context.TraceContext;
+
+            using (var scope11 = tracer.StartActive("1.1"))
+            {
+                span11 = scope11.Span;
+            }
+
+            using (var scope12 = tracer.StartActive("1.2"))
+            {
+                span12 = scope12.Span;
+
+                using (var scope121 = tracer.StartActive("1.2.1"))
+                {
+                    span121 = scope121.Span;
+                }
+            }
+
+            // send the finished spans as one trace chunk
+            traceContext.WriteClosedSpans();
+        }
+
+        // send the remaining spans as another trace chunk
+        await tracer.FlushAsync();
+        var traceChunks = _testApi.Wait();
+
+        // expected chunks:
+        // [ 1.1, 1.2, 1.2.1 ]
+        // [ 1 ]
+
+        traceChunks.Should().HaveCount(2);    // 2 trace chunks
+        traceChunks[0].Should().HaveCount(3); // 3 spans
+        traceChunks[1].Should().HaveCount(1); // 1 span
+
+        // chunk 0, both orphan spans should have all aas tags
+        traceChunks[0]
+           .Where(s => s.SpanId == span11.SpanId || s.SpanId == span12.SpanId)
+           .Should()
+           .HaveCount(2)
+           .And.OnlyContain(s => s.ParentId == span1.SpanId)
+            // I don't know why Assume doesn't work in that case but using SatisfyRespectively does the trick
+           .And.SatisfyRespectively(s => AssertLocalRootSpan(s), s => AssertLocalRootSpan(s));
```

Comment:
Small suggestion for readability:
```suggestion
           .And.SatisfyRespectively(firstSpan => AssertLocalRootSpan(firstSpan), secondSpan => AssertLocalRootSpan(secondSpan));
```

Reply:
Or it looks like you could do the following:
```suggestion
           .And.AllSatisfy(s => AssertLocalRootSpan(s));
```

Reply:
Fixed with the first suggestion in 00cd70fc4b7871c3a9846286093009db3706ddfd.